syntax = "proto3";
import "hyweb.proto";

message Data {
    float score = 1;
    int32 index = 2;
}

message SolverWithConfig {
    hyweb.Service definition = 1;
    map<string, hyweb.Any> enviroment_variables = 2;
}

message DataSetInstance {
    SolverWithConfig solver = 1;
    map<string, Data> data = 2;
    string hash = 3; // identifica el Solver con una configuracion determinada.( hash.value.hex() )
}

message DataSet {
    repeated DataSetInstance data = 1;
}